examples/gcoap/gcoap_cli.c

```diff
@@ -93,11 +96,38 @@ static void _resp_handler(unsigned req_state, coap_pkt_t* pdu,
  */
 static ssize_t _stats_handler(coap_pkt_t* pdu, uint8_t *buf, size_t len)
 {
-    gcoap_resp_init(pdu, buf, len, COAP_CODE_CONTENT);
+    /* read coap method type in packet */
+    unsigned method_flag = coap_method2flag(coap_get_code_detail(pdu));
+
+    switch(method_flag) {
+        case COAP_GET:
+            gcoap_resp_init(pdu, buf, len, COAP_CODE_CONTENT);
+
+            /* write the response buffer with the request count value */
+            size_t payload_len = fmt_u16_dec((char *)pdu->payload, req_count);
+
+            return gcoap_finish(pdu, payload_len, COAP_FORMAT_TEXT);
+        case COAP_PUT:
+        case COAP_POST:
+        {
+            gcoap_resp_init(pdu, buf, len, COAP_CODE_CHANGED);
+
+            /* convert the payload to an integer and update the request count value */
+            char payload[16] = { 0 };
+            memcpy(payload, (char*)pdu->payload, pdu->payload_len);
+            req_count = strtol(payload, NULL, 10);
```

Comment:
Use strtoul() and cast the right-hand result to uint16_t.

Reply:
In fact, the PUT handler was not working. I fixed it.